Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -88,8 +148,28 @@ public static int[] InsertLast(int[] array, int number)
         /// <returns>A new array with element inserted at a given index. If an array is empty or null, returns input array.</returns>
         public static int[] InsertAt(int[] array, int number, int index)
         {
-            // ToDo: implement.
-            return array;
+            if (array == null || array.Length == 0)
+            {
+                return array;
+            }
+
+            int[] tempArr = new int[array.Length + 1];
+            int oldArrayIndex = 0;
+            for (int i = 0; i < tempArr.Length; i++)
+            {
+                if (i == index)
+                {
+                    tempArr[i] = number;
+                    i++;
+                    if (index == array.Length)
+                    {
```

Comment:
This should be in front of the for loop, shouldn't it?

Reply:
If you are talking about the if statement that returns the array inside the if statement that checks if the new array is at the requested index this is my reasoning for doing it. 
This check if the spot to insert was at the end of the array. If it is then there is no reason to proceed and just returns there. If it isn't then there would be more of the array to shift over after the insert. If what you are pointing put here is that second if statement.

Reply:
Yeah I just figured that I misunderstood your code. Sorry for that! It's fine. :)